frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -154,12 +158,86 @@ private IGitRepositorySelectionProvider getGitRepositorySelectionProvider() {
 
   private void subscribeToGitRepositoryFilesChanges() {
     Topic<GitRepositoryChangeListener> topic = GitRepository.GIT_REPO_CHANGE;
-    GitRepositoryChangeListener listener = repository -> queueRepositoryUpdateAndModelRefresh();
+    GitRepositoryChangeListener listener = repository -> {
+      trackCurrentBranchChange(repository);
+      queueRepositoryUpdateAndModelRefresh();
+    };
+
     val messageBusConnection = project.getMessageBus().connect();
     messageBusConnection.subscribe(topic, listener);
     Disposer.register(project, messageBusConnection);
   }
 
+  private Notification getNotificationAboutNonManagedBranch(String branchName) {
```

Comment:
for now it works different from what we described in the related issue (my mistake - I wanted to get smth working first and then improve it). 
notification happens first, then on `slide in` action parent is being inferred.

probably the better approach is the initial one (from the related issue) - to have the inferred branch name in the notification. Then it could look like

```
Branch 'foo' is not managed by Git Machete.
Consider sliding in 'foo' under inferred parent ('master') 
or edititing the machete file manually.
<Slide In> <Open Machete file> 
```

what do you think?

Reply:
+1, let's show the inferred parent branch indeed (e.g. `Slide in under develop` instead of `Slide in`)